src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs

```diff
@@ -146,7 +146,7 @@ public void Equals_Object_ReturnsExpected(GCHandle handle, object other, bool ex
             finally
             {
                 handle.Free();
-                if (other is GCHandle otherHandle)
+                if (other is GCHandle otherHandle && !expected)
```

Comment:
This will work in this case I think, but I'm not sure it's the right fix.

I think the issue is that `handle` and `otherHandle` are the same instance:

```
if (other is GCHandle otherHandle && !ReferenceEquals(handle, otherHandle))
```

I'm not sure how important this is, as I don't know if there are situations where `GCHandle` can be equal to a different instance